src/main/java/org/peergos/Server.java

```diff
@@ -117,6 +119,25 @@ public Server(Args args) throws Exception {
             throw new IllegalStateException("No connected peers!");
         dht.bootstrap(node);
 
+        new Thread(() -> {
+            while (true) {
+                try {
+                    List<Cid> refs = blockStore.refs().join();
+                    PeerAddresses ourAddrs = PeerAddresses.fromHost(node);
+                    List<CompletableFuture<Void>> published = refs.stream().parallel()
+                            .map(ref -> dht.provideBlock(ref, node, ourAddrs))
+                            .collect(Collectors.toList());
+                    for (CompletableFuture<Void> fut : published) {
+                        fut.join();
+                    }
+                    long reprovideInterval = 22 * 3600_000L;
+                    Thread.sleep(reprovideInterval);
+                } catch (Exception e) {
```

Comment:
make it Throwable to catch Error and Error subclasses as well

Reply:
done